File: C++/CodeForces/team-training.cpp
#include <iostream>
using namespace std;

int main(){
    int xp, nb, groups = 0;
    cin >> xp >> nb;

    while(xp + nb >= 3 && nb > 0 && xp > 0){
        if(xp > nb){
            xp -= 2;
            nb -=1;
            groups++;
        } else {
            xp -= 1;
            nb -= 2;
            groups++;
        }
    }
    cout << groups;

    return 0;
}

/* 519C - A and B and Team Training
https://codeforces.com/problemset/problem/519/C */

File: Codeforces/team-training/team-training.py
/* Problem statement: https://codeforces.com/problemset/problem/519/C */
xp, nb = map(int, input().split())
groups = 0

while xp + nb >= 3 and nb > 0 and xp > 0:
    if xp > nb:
        xp -= 2
        nb -= 1
        groups += 1
    else:
        xp -= 1
        nb -= 2
        groups += 1

print(groups)

